chained/water_version_control.go

```diff
@@ -0,0 +1,274 @@
+package chained
+
+import (
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/getlantern/flashlight/v7/proxied"
+	"github.com/gocarina/gocsv"
+)
+
+type versionControl struct {
+	// contain all wasm files available locally
+	// key is something like: protocol.version
+	wasmFilesAvailable map[string]wasmInfo
+	dir                string
+
+	// history of the last time the WASM file was loaded
+	history      []history
+	historyMutex *sync.Mutex
+}
+
+type history struct {
+	Transport      string    `csv:"transport"`
+	LastTimeLoaded time.Time `csv:"last_time_loaded"`
+}
+
+type wasmInfo struct {
+	version   string
+	protocol  string
+	builtWith string
+	path      string
+}
+
+type VersionControl interface {
```

Comment:
There is only one implementation of this interface, so let's do away with it and just use the implementation directly.